src/corecel/math/ArrayOperators.hh

```diff
@@ -53,9 +55,9 @@ namespace celeritas
         return (result TOKEN## = y);                                       \
     }                                                                      \
                                                                            \
-    template<class T, size_type N>                                         \
+    template<class T, size_type N, class T2 = std::remove_cv_t<T>>         \
     inline CELER_FUNCTION Array<T, N> operator TOKEN(Array<T, N> const& x, \
-                                                     T const& y)           \
+                                                     T2 const& y)          \
```

Comment:
This is just an implementation detail I think it's fine, but where did this show up? (What was the error that required this?)

Reply:
it showed up [here](https://github.com/celeritas-project/celeritas/blob/88237d15dbd170a7cf4a7fcd0939bfd985d29e9e/src/orange/surf/detail/SurfaceTransformer.cc#L195-L196) when make_array was returning an `array<T const>` if passed a `LdgSpan<T const>`. Since we make a copy of each element and I changed make_array in this PR to remove the constness of T this is no longer needed. Keeping this in might still make sense, `x` and `y` don't require the same const qualifiers for `T`.

Reply:
Shouldn't this return `Array<T2, N>`?  Should we prohibit `Array<T const>`?

Reply:
It is already indirectly prohibited because the operations defined by `CELER_DEFINE_ARRAY_ASSIGN` only support non-const